iBudget/utils/validators.py

```diff
@@ -118,6 +119,31 @@ def input_spending_registration_validate(data):
     except (ValidationError, AttributeError):
         return False
 
+def input_fund_registration_validate(data):
+    """validate data.
+        Args:
+            data (dict): contain fund, value
+        Returns:
+            bool: The return value. True is data valid, else False.list([1,2,3])
+    """
+    if not set(data.keys()).difference(SET_KEYS_FUND_GOAL):
+        return False
+    try:
+        data['fund'] = int(data['fund'])
+        data['value'] = Decimal(data['value'])
+        return True
+    except (ValidationError, AttributeError):
+        return False
+
+def date_range_validate(data):
+    """
+    Function that provides data range validation
+    :param data: start_date, finish_date
+    :return: Raise validation error when the starting date is greater than the final one
+    """
+    if data["start_date"] > data["finish_date"]:
```

Comment:
Could user press Save button on the page when start date is after finish date?

Reply:
Yes, but then the validator will work and it will give an error.

Reply:
For the future: maybe it will be good to implement validation on front-end too.

Reply:
he's already there